### ISSUE TITLE
<Component> was created without expected prop

### ISSUE THREAD

**Issue by secondl1ght** [sveltejs/eslint-plugin-svelte#483]
### Motivation

Maybe I am missing it somewhere but I would like this Warning that shows up in the console to be available as a rule. This would be useful for finding missing props.

### Description

The same rule that shows up in the DevTools console.

### Examples

```svelte
<script>
export let title
</script>

<!-- ✓ GOOD -->
<Foo title='Hello' />

<!-- ✗ BAD -->
<Foo />
```

### Additional comments

Thanks

**Comment by ota-meshi** [comment]
I don't quite understand the new rules you are requesting.
Can you explain the specs to check?

**Comment by secondl1ght** [comment]
If a component is created without an expected prop then a warning appears.

Here is a screenshot of an example:
![image](https://github.com/sveltejs/eslint-plugin-svelte/assets/85003930/5feb20dc-b2bf-4cdc-8d32-bee89cb636f8)

In the OP example you can see that the `Foo` component is missing the `title` prop in the BAD example. This would result in the Warning. The GOOD example has the expected prop so no Warning would appear.

This is useful because you might have forgot to add the prop to the component, so this will help catch those cases. If a missing prop is intentional, then that means it is an optional prop and should be declared like this to avoid the Warning:

```svelte
export let title = undefined
```

**Comment by ota-meshi** [comment]
Ah. I didn't realize before that the `export let title` was for the `Foo` component.
However, implementing that rule is difficult due to ESLint's mechanics. ESLint rules only refer to and check single files. As such, it cannot know the contents of other files it is referencing.
If you have any good ideas please let me know.

**Comment by secondl1ght** [comment]
I thought ESLint can reference other files? For example it's able to determine if an imported variable or file exists. If the path is not found it or variable is not exported, then it will result in an error.
